### PR TITLE
Fix: Various UI Improvements, Fixes and Backend Refactoring

### PR DESCRIPTION
**Pull request by scott-ray-wilson**
# Description 📣

This PR adds AWS region slugs to all integration dropdowns, fixes dropdown overflow, adds explanatory tooltips to the secrets table resource counter, moves the dashboard router to v1, and improves handling of querystring boolean values for dashboard endpoints.

## Type ✨

- [x] Improvements
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Comment by scott-ray-wilson** [comment]
@akhilmhdh this is not a publically exposed endpoint so the breaking API change is fine correct?

**Comment by akhilmhdh** [comment]
@scott-ray-wilson Yes it's fine

File: backend/src/server/routes/v1/dashboard-router.ts [hunk 1]
```diff
@@ -17,6 +17,20 @@ import { AuthMode } from "@app/services/auth/auth-type";
 import { SecretsOrderBy } from "@app/services/secret/secret-types";
 import { PostHogEventTypes } from "@app/services/telemetry/telemetry-types";
 
+// handle querystring boolean values
+const booleanSchema = z
+  .union([z.boolean(), z.string().trim()])
+  .transform((value) => {
+    if (typeof value === "string") {
+      // ie if not empty, 0 or false, return true
+      return !!value && Number(value) !== 0 && value.toLowerCase() !== "false";
```
Comment (akhilmhdh) [inline backend/src/server/routes/v1/dashboard-router.ts:26]: This should be an OR gate right?

Also `!!value` => Boolean(value)

**Review by akhilmhdh — CHANGES_REQUESTED** [review]

**Comment by akhilmhdh** [comment]
I'll test this with an integration today and get back to you interms of UI

File: backend/src/server/routes/v1/dashboard-router.ts (hunk 1 above)
Comment (scott-ray-wilson) [reply backend/src/server/routes/v1/dashboard-router.ts:26]: Will swap to `Boolean(value)` but why would should this be an OR gate? 

it's essentially saying return `true` only if not "" and "0" and "false",  which all of these should evaluate true for it to be `true`?

**Comment by scott-ray-wilson** [comment]
@akhilmhdh bump

**Review by akhilmhdh — APPROVED** [review]

**Pull request merged**